k8s/src/test/scala/io/buoyant/k8s/EndpointsNamerTest.scala

```diff
@@ -1018,4 +1042,82 @@ class EndpointsNamerTest extends FunSuite with Awaits {
       assert(state == Activity.Ok(NameTree.Neg))
     }
   }
+
+  test("routes to new endpoints after scale down followed by scale up") {
+    // linkerd #1626 reproduction
+    @volatile var writer: Writer = null
+
+    val service = Service.mk[Request, Response] {
+      case req if req.uri == "/api/v1/namespaces/liza1626/endpoints/world-v1" =>
+        val rsp = Response()
+        rsp.content = Rsps1626.Init
+        Future.value(rsp)
+      case req if req.uri.startsWith("/api/v1/watch/namespaces/liza1626/endpoints/world-v1") =>
+        val rsp = Response()
+        rsp.setChunked(true)
+
+        writer = rsp.writer
+//        val init = writer.write(Rsps1626.InitWatch)
+//        init.onSuccess{ _ => val _ = didInit.setDone() }
+
+//        init before {
+//          doScaleDown before {
+//            val scaleDown = writer.write(Rsps1626.BeforeScaleDown) before
+//              writer.write(Rsps1626.ScaleDown)
+//            scaleDown onSuccess { _ => val _ = didScaleDown.setDone() }
+//            scaleDown before {
+//              doScaleUp before
+//                writer.write(Rsps1626.ScaleUp1) before
+//                writer.write(Rsps1626.ScaleUp2) before
+//                writer.write(Rsps1626.ScaleUp3)
+//            }
+//          }
+//        }
+
+        Future.value(rsp)
+    }
+    val api = v1.Api(service)
+    val timer = new MockTimer
+    val namer = new MultiNsNamer(Path.read("/test"), None, api.withNamespace, Stream.continually(1.millis))(timer)
+
+    @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
+
+    val activity = namer.lookup(Path.read("/liza1626/http/world-v1"))
+
+    val _ = activity.states respond { s =>
+      state = s
+    }
+
+    def addrs = state match {
+      case Activity.Ok(NameTree.Leaf(bound: Name.Bound)) =>
+        bound.addr.sample() match {
```

Comment:
bound.addr is the Var whose identity you want to make sure is preserved.

Reply:
yeah, added a test for this (183d08834fe75ed9832ac8da3d1f7d91aa06897a) but evidentially forgot to push it up? anyway, rectified that.